src/org/exist/storage/DBBroker.java

```diff
@@ -123,11 +124,27 @@ public DBBroker(final BrokerPool pool, final Configuration config) {
             caseSensitive = temp.booleanValue();
         }
         this.pool = pool;
-        initIndexModules();
     }
 
-    public void initIndexModules() {
-        indexController = new IndexController(this);
+    /**
+     * Prepares the broker for (re-)use,
+     * when (re-)leased from BrokerPool.
+     */
+    public void prepare() {
+        /**
+         * Index modules should always be re-loaded in case
+         * {@link org.exist.indexing.IndexManager#registerIndex(Index)} or
+         * {@link org.exist.indexing.IndexManager#unregisterIndex(Index)}
+         * has been called since the previous lease of this broker.
+         */
+        loadIndexModules();
+    }
+
+    /**
+     * Loads the index modules via an IndexController
+     */
+    protected void loadIndexModules() {
+        indexController.setIfExpiredOrNull(getBrokerPool().getIndexManager().getConfigurationTimestamp(), () -> new IndexController(this));
```

Comment:
this looks as over-engineering to me because broker have `single thread` contract